<launch>
  <!-- Load URDF from file -->
  <param name="robot_description" textfile="$(find my_humanoid_robot)/urdf/my_humanoid_robot.urdf" />

  <!-- Start Robot State Publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"/>

  <!-- Joint State Publisher GUI -->
  <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" output="screen"/>

  <!-- Start Forward Kinematics Node -->
  <node name="forward_kinematics_node" pkg="my_humanoid_robot" type="forward_kinematics.py" output="screen"/>

  <!-- Start RViz -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find my_humanoid_robot)/config/my_humanoid_robot_config.rviz" output="screen"/>
</launch>
